config.c

```diff
@@ -2133,7 +2137,28 @@ static int readConfigFile(const char *configFile, struct logInfo *defConfig)
                                 }
                             }
 
-                            free(ld);
+                            /* Check if olddir and its parents of this log has safe permissions */
+                            r = checkIsSecureDir(dirName, newlog);
```

Comment:
`r` is defined only with `HAVE_MADVISE`.

Reply:
thanks, fixed